relayer/src/link.rs

```diff
@@ -856,13 +895,39 @@ impl Link {
 
     pub fn relay(&mut self) -> Result<(), LinkError> {
         println!("relaying packets on {:#?}", self.a_to_b.channel);
+
+        // First time in the loop the old events emitted before the link started the event
+        // subscription need to be processed.
+        let mut clear_events_a = true;
+        let mut clear_events_b = true;
+
         loop {
             if self.is_closed()? {
                 println!("channel is closed, exiting");
                 return Ok(());
             }
-            self.a_to_b.relay_from_events()?;
-            self.b_to_a.relay_from_events()?;
+
+            match self.a_to_b.relay_from_events(clear_events_a) {
+                Ok(()) => {
+                    clear_events_a = false;
+                    Ok(())
+                }
+                Err(e) => match e {
+                    LinkError::OldPacketClearingPending => Ok(()),
+                    _ => Err(e),
+                },
+            }?;
+
+            match self.b_to_a.relay_from_events(clear_events_b) {
+                Ok(()) => {
+                    clear_events_b = false;
+                    Ok(())
+                }
+                Err(e) => match e {
+                    LinkError::OldPacketClearingPending => Ok(()),
+                    _ => Err(e),
+                },
+            }?;
```

Comment:
I find this slightly more readable as it decreases the indentation a bit, but feel free to ignore.

```suggestion
            match self.a_to_b.relay_from_events(clear_events_a) {
                Ok(()) => {
                    clear_events_a = false;
                    Ok(())
                }
                Err(LinkError::OldPacketClearingPending) => Ok(()),
                Err(e) => Err(e),
            }?;

            match self.b_to_a.relay_from_events(clear_events_b) {
                Ok(()) => {
                    clear_events_b = false;
                    Ok(())
                }
                Err(LinkError::OldPacketClearingPending) => Ok(()),
                Err(e) => Err(e),
            }?;
```